boards/arm64/a64/pinephone/Kconfig

```diff
@@ -17,4 +17,10 @@ config PINEPHONE_LCD
 	---help---
 		Select to enable support for LCD Display.
 
+config PINEPHONE_MODEM
+	bool "LTE Modem"
+	default n
+	---help---
+		Select to enable support for Quectel EG25-G LTE Modem.
```

Comment:
Can we abstract a modem interface? So, the SoC can support the different modem, and the modem can support different SoC.

Reply:
Sorry I need to do some research to see how the SoC can support other modems. I don't believe NuttX has a modem interface today?

Reply:
There are some in drivers/modem, but I am not sure where the common driver interface exists or not. It will be good additional to unify the interface between apps and all possible modem hardware.

Reply:
Thanks let me take a look again. The last time I searched, I found AT Commands referenced in 2 places:
- [uart_lorawan_layer.c](https://github.com/apache/nuttx-apps/blob/master/examples/tcp_ipc_server/uart_lorawan_layer.c#L262-L274)
- [esp8266.c](https://github.com/apache/nuttx-apps/blob/master/netutils/esp8266/esp8266.c#L1573-L1582)

Reply:
@lupyuen I don't think there is a standard command driver for modems, because normally they are hardware dependent. Also I don't think these examples you cited are the right way to support Direct-IP modems on NuttX, Direct-IP modems need to use USRSOCK to integrated correctly with NuttX.

If this modem supports PPP protocol then this is a good way to go (even been a legacy protocol it is well supported on NuttX). I friend of mine (@duduita) is using BG770A and BG95 (Cat-M1 and NB-IoT) with PPP on NuttX and it works fine.

If this Quectel EG25-G is also connected to the USB and has support to MBIM protocol (MBIM and QMI are the most common protocols used on 4G modems) you can use the drivers/usbhost/usbhost_cdcmbim.c with it.